mongo/src/main/java/com/redhat/lightblue/mongo/crud/BasicDocSaver.java

```diff
@@ -358,10 +358,13 @@ private void updateDocs(CRUDOperationContext ctx,
                         doc.inputDoc.setCRUDOperationPerformed(CRUDOperation.UPDATE);
                     }
                     try {
-                        Map<Integer,Error> errorMap=upd.commit();
-                        for(Map.Entry<Integer,Error> entry:errorMap.entrySet()) {
+                        BatchUpdate.CommitInfo ci=upd.commit();
+                        for(Map.Entry<Integer,Error> entry:ci.errors.entrySet()) {
                             updateAttemptList.get(entry.getKey()).inputDoc.addError(entry.getValue());
-                        }                        
+                        }
+                        for(Integer i:ci.lostDocs) {
+                            updateAttemptList.get(i).inputDoc.addError(Error.get("update",MongoCrudConstants.ERR_DOC_NO_LONGER_AVAILABLE,""));
```

Comment:
Should these really be considered errors? We're just serializing operations that occurred concurrently. It's not an error that some no longer match the query, it's what should happen in order to establish the predictable happens-before relationship that ensures the atomicity we expect.

For a real world example, if I atomically find-and-update several documents only if they match a timestamp, and update that timestamp, two of these requests interweaving should produce the same results as two in serial. If they interweave and one fails, that's not the same result. If the concurrently changed docs that no longer match the timestamp are dropped, that does give us the same result. It's correct and expected way to recover from an error, not an error itself as I understand it.